### DISCUSSION TITLE
When can we expect tempo native search enabled?

### DISCUSSION THREAD

**Discussion by mvkrishna86** [grafana/tempo#1123] [General]
Any probable time for the tempo native search to be fully production ready?: https://grafana.com/docs/tempo/latest/configuration/#search

**Comment by joe-elliott** [comment]
The current goal is to have this production ready and available in Grafana Cloud by end of Quarter (early next year). This will probably include a documented method of using Google Cloud Functions to help parralelize the work of searching the backend. We also intend to provide methods of using serverless technologies for other major cloud providers over time.

**Reply by mvkrishna86** [reply]
Will that be available only in Grafana Cloud or standalone installation also?

**Reply by joe-elliott** [reply]
It will be in OSS Tempo so anyone will be able to use it.

**Reply by mvkrishna86** [reply]
1.3.0 is released which has the full backend search, but I dont see any API changes in the release doc. So can I assume the existing grafana will work with the experimental search feature enabled?

**Reply by yvrhdn** [reply]
The Tempo search API now accepts 2 new optional parameter to do a full backend search ([the `start` and `end` parameter as described here](https://grafana.com/docs/tempo/latest/api_docs/#search)). Grafana has to send these, otherwise Tempo will do a recent search only. The difference between both search methods is described over here: https://grafana.com/docs/tempo/latest/getting-started/tempo-in-grafana/#tempo-search

Finally, if you are running Tempo yourself, you will probably want to take a look here as well: https://grafana.com/docs/tempo/latest/operations/backend_search/

Hope that helps!
